homeassistant/components/xbox/config_flow.py

```diff
@@ -0,0 +1,29 @@
+"""Config flow for xbox."""
+import logging
+
+from homeassistant import config_entries
+from homeassistant.helpers import config_entry_oauth2_flow
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class OAuth2FlowHandler(
```

Comment:
Can we set a unique id in the flow to guard from duplicates when setting up more than one config entry?

Reply:
Yes, can it be the OAuth2 user variable? If so, should we also update the skeleton to use that automatically?

Reply:
Never mind, that’s not in the spec so may just be an additional variable in the Microsoft token response. I’ll try logging into a different account to verify uniqueness.

Reply:
Do you mean the `username`, part of the oauth2 spec?

Different integrations may have a different structure of their data, eg multiple devices under one account, and may wish to have a different structure of identifying the config entries. We could have a default unique id as long as the integration can override it. At the same time there are variations of oauth2 implementations that don't quite follow the spec.